src/AzureAppConfigurationImpl.ts

```diff
@@ -176,35 +175,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         return {
             requestTracingEnabled: this.#requestTracingEnabled,
             initialLoadCompleted: this.#isInitialLoadCompleted,
-            appConfigOptions: this.#options
+            appConfigOptions: this.#options,
+            isFailoverRequest: this.#isFailoverRequest
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute: (client: AppConfigurationClient) => Promise<any>): Promise<any> {
+        const clientWrappers = await this.#clientManager.getClients();
+        if (clientWrappers.length === 0) {
+            this.#clientManager.refreshClients();
+            console.warn("Refresh skipped because no endpoint is accessible.");

```

Comment:
This is from .NET provider. But it doesn't make any sense. We actually do the refresh because no endpoint is accessible.

Reply:
https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L232

Reply:
In the .NET provider, the refresh means kv refresh operation. The .net implementation is different from here. The logic of getClients is not included in the executeWithFailoverPolicy. The message is specific for the `Refresh` method of provider.

Reply:
If we we want to have the same behavior as .NET, we should return here and do nothing. Here the load and refresh scenarios are different. We can tolerant refresh doing nothing if there is no client available. No exception should be thrown, just log warning is fine. The load scenario is more important, we should make sure that configuration is pulled and if fail, throw exception.

Reply:
discussed offline, log warning in refresh()